tools/ola_trigger/contrib/elgato_key_light.conf

```
@@ -0,0 +1,127 @@
+2       166-171  `curl '-s' '-o /dev/null' '-k' '-H' '"Accept: application/json"' '-X' 'PUT' '--data' '{"numberOfLights":1,"lights":[{"on":1,"temperature":178}]}' 'http://${hostname}/elgato/lights'`
+2       172-177  `curl '-s' '-o /dev/null' '-k' '-H' '"Accept: application/json"' '-X' 'PUT' '--data' '{"numberOfLights":1,"lights":[{"on":1,"temperature":175}]}' 'http://${hostname}/elgato/lights'`
+2       178-183  `curl '-s' '-o /dev/null' '-k' '-H' '"Accept: application/json"' '-X' 'PUT' '--data' '{"numberOfLights":1,"lights":[{"on":1,"temperature":172}]}' 'http://${hostname}/elgato/lights'`
+2       184-189  `curl '-s' '-o /dev/null' '-k' '-H' '"Accept: application/json"' '-X' 'PUT' '--data' '{"numberOfLights":1,"lights":[{"on":1,"temperature":169}]}' 'http://${hostname}/elgato/lights'`
+2       190-195  `curl '-s' '-o /dev/null' '-k' '-H' '"Accept: application/json"' '-X' 'PUT' '--data' '{"numberOfLights":1,"lights":[{"on":1,"temperature":166}]}' 'http://${hostname}/elgato/lights'`
+2       196-201  `curl '-s' '-o /dev/null' '-k' '-H' '"Accept: application/json"' '-X' 'PUT' '--data' '{"numberOfLights":1,"lights":[{"on":1,"temperature":163}]}' 'http://${hostname}/elgato/lights'`
+2       202-207  `curl '-s' '-o /dev/null' '-k' '-H' '"Accept: application/json"' '-X' 'PUT' '--data' '{"numberOfLights":1,"lights":[{"on":1,"temperature":161}]}' 'http://${hostname}/elgato/lights'`
+2       208-213  `curl '-s' '-o /dev/null' '-k' '-H' '"Accept: application/json"' '-X' 'PUT' '--data' '{"numberOfLights":1,"lights":[{"on":1,"temperature":158}]}' 'http://${hostname}/elgato/lights'`
+2       214-219  `curl '-s' '-o /dev/null' '-k' '-H' '"Accept: application/json"' '-X' 'PUT' '--data' '{"numberOfLights":1,"lights":[{"on":1,"temperature":156}]}' 'http://${hostname}/elgato/lights'`
+2       220-225  `curl '-s' '-o /dev/null' '-k' '-H' '"Accept: application/json"' '-X' 'PUT' '--data' '{"numberOfLights":1,"lights":[{"on":1,"temperature":154}]}' 'http://${hostname}/elgato/lights'`
+2       226-231  `curl '-s' '-o /dev/null' '-k' '-H' '"Accept: application/json"' '-X' 'PUT' '--data' '{"numberOfLights":1,"lights":[{"on":1,"temperature":152}]}' 'http://${hostname}/elgato/lights'`
+2       232-237  `curl '-s' '-o /dev/null' '-k' '-H' '"Accept: application/json"' '-X' 'PUT' '--data' '{"numberOfLights":1,"lights":[{"on":1,"temperature":149}]}' 'http://${hostname}/elgato/lights'`
+2       238-243  `curl '-s' '-o /dev/null' '-k' '-H' '"Accept: application/json"' '-X' 'PUT' '--data' '{"numberOfLights":1,"lights":[{"on":1,"temperature":147}]}' 'http://${hostname}/elgato/lights'`
+2       244-249  `curl '-s' '-o /dev/null' '-k' '-H' '"Accept: application/json"' '-X' 'PUT' '--data' '{"numberOfLights":1,"lights":[{"on":1,"temperature":145}]}' 'http://${hostname}/elgato/lights'`
+2       250-255  `curl '-s' '-o /dev/null' '-k' '-H' '"Accept: application/json"' '-X' 'PUT' '--data' '{"numberOfLights":1,"lights":[{"on":1,"temperature":143}]}' 'http://${hostname}/elgato/lights'`
```

I think you should be able to simplify/de-duplicate this, if you look at:
https://github.com/OpenLightingProject/ola/blob/master/tools/ola_trigger/contrib/mac_itunes.conf

Something like this (entirely untested):
```suggestion
2       4-9      ct='344'
2       10-15    ct='335'
2       16-21    ct='323'
...
2       250-255    ct='143'
2       4-255  `curl '-s' '-o /dev/null' '-k' '-H' '"Accept: application/json"' '-X' 'PUT' '--data' '{"numberOfLights":1,"lights":[{"on":1,"temperature":${ct}}]}' 'http://${hostname}/elgato/lights'`
```

I think the fact you're already doing it for hostname should mean it just works! :crossed_fingers: Unless it doesn't like the duplicate entry for the channel and values?

You could probably set both and then fire once to avoid two curl calls too, which might help responsiveness?

---

The only thing to consider with one curl call would be if you don't send enough channels, it might not do anything, but that probably matches lots of real lights TBH.

---

I probably hadn't really thought of that, will try this solution :thumbsup:

---

It does not like the duplicate entry for the channel and values:

```
# test.conf
hostname='192.168.0.1:9123'
ct=''

2       4-9      ct='344'
2       10-15    ct='335'
2       16-21    ct='323'

2       4-21     `echo 'test ct: ${ct} hostname: ${hostname}'`
```
```
$ ola_trigger --validate test.conf
tools/ola_trigger/Action.cpp:405: Interval [4, 21] overlaps [4, 9]
tools/ola_trigger/ParserActions.cpp:156: Line 19: value [4, 21] collides with existing values.
```

---

> It does not like the duplicate entry for the channel and values:

Try this instead:
```
# test.conf
hostname='192.168.0.1:9123'
ct=''

2       4-9      ct='344'
2       10-15    ct='335'
2       16-21    ct='323'

2       %        `echo 'test ct: ${ct} hostname: ${hostname}'`
```

I'm not sure what will happen when you send values 1-3 as currently set...

I think this should work, it currently passes the validation at least, so we should probably either fix the behaviour or the validation if it doesn't...

---

Blank ct output on value 4-21. Value 1-3, 22-255 response ct 323 on channel 2

---

So if you see this:
```
1       4-9      `echo 'Value 4-9'`
1       10-15    `echo 'Value 10-15'`
1       16-21    `echo 'Value 16-21'`
1       %        `echo 'Generic value'`
```

I think it's working in a sensible and effective way, although not very useful for this particular purpose.

Unfortunately I just tried this:
```
# test.conf
1       4-9      `echo 'Value 4-9'`
1       10-15    `echo 'Value 10-15'`
1       16-21    `echo 'Value 16-21'`
1       %        `echo 'Generic value'`
2       %        `echo 'Always send'`
```

But always send is only fired when slot two changes value.

So for this PR/without other changes, I think what you had is pretty good and we can't change much (aside from possibly the on/off thing mentioned elsewhere).

I guess this is another thing for #1848 though, there could be some sort of pre and post processing options where it would run the given command before or after and slot based processing if that slot had changed. So say:
```
# test.conf
1       pre      `echo 'Run before'`
1       4-9      `echo 'Value 4-9'`
1       10-15    `echo 'Value 10-15'`
1       16-21    `echo 'Value 16-21'`
1       %        `echo 'Generic value'`
1       post     `echo 'Run after'`
2       %        `echo 'Always send'`
```

---

Sorry for the late reply but I've been sick off and on lately. I will continue with this when I recover. Will also order a Key Light to verify it works properly.

---

> Sorry for the late reply but I've been sick off and on lately. I will continue with this when I recover.

Sorry to hear that @gobo-ws . Obviously no rush whatsoever, I hope you're feeling better soon!

> Will also order a Key Light to verify it works properly.

I'm pretty certain I can still get access to one, so please don't buy one just for testing purposes (although they do look fun)!

---

I ordered an Elgato Key Light (will not keep it though) and made some tests.


Elgato Key Light dimming range starts at 3 % in the Elgato Control Center app but when using the API the light lights up already at 2 percent.

Sending 1 % does not generate any error from the API but I can't see any visible light output.
If you set the value to zero by using the API the light will turn off (it will change the light state on to zero)
So I think we keep the 0-100 % range.

Don't know if it applies to all models though.

I have a problem to get a soft dimmer level with OLA when using the fader in the web interface with this trigger file.
Don't remember having this issue before. But I think it works a little bit better when using the 0-100 range instead of scaling the input from 0-255 to 0-100.
Also made some tests by removing sending the on command on every brightness request but it made no difference.

---

Any comments @peternewman?

---

> Elgato Key Light dimming range starts at 3 % in the Elgato Control Center app but when using the API the light lights up already at 2 percent.

So 2% is when it first turns on?

> Sending 1 % does not generate any error from the API but I can't see any visible light output. If you set the value to zero by using the API the light will turn off (it will change the light state on to zero) So I think we keep the 0-100 % range.

Yeah that makes sense in terms of the final output.

> Don't know if it applies to all models though.

Without a published API I think we just have to go with what works for one unit and wait for people to find issues!

> I have a problem to get a soft dimmer level with OLA when using the fader in the web interface with this trigger file. Don't remember having this issue before. But I think it works a little bit better when using the 0-100 range instead of scaling the input from 0-255 to 0-100. Also made some tests by removing sending the on command on every brightness request but it made no difference.

Have you tried logging what its sending to check there's nothing wrong with the maths or similar? TBH this feels more like its an issue with the web interface's fader than anything else to me.

Have you tried using say ola_recorder or one of the examples to play back a fade from 0-255 over a certain amount of time? Then you could switch between the two configs and see how it behaves (aside from speeding up when going over a shorter range of inputs).

My gut feeling is still that the benefit of it behaving just like any other fixture and running over the whole range cancels out any potential minor issues with our web UI.

> Running firmware: 1.0.3 (217) Did not want to update the firmware since I am not keeping the device.

Yeah fair enough.